proxy_test.go

```diff
@@ -69,10 +69,13 @@ func hostportnum(u string) (string, int) {
 	return host, port
 }
 
-func init() {
+func doInit() {
 	srv = httptest.NewServer(mux())
 	listen = hostport(srv.URL)
 }
+func init() {
+	doInit()
```

Comment:
Don't get this magic with `doInit`. Why we need to launch this initialization only is some tests? Seems to complicate things.